pilot/pkg/model/push_context.go

```diff
@@ -985,6 +1002,32 @@ func (ps *PushContext) initDestinationRules(env *Environment) error {
 	return nil
 }
 
+// AuthPolicyForProxy returns the matching auth policy for a given proxy
+func (ps *PushContext) AuthPolicyForProxy(service *Service, port int, proxy *Proxy) *authn.Policy {
```

Comment:
I am a bit confuse here. Is proxy is one instance of the service?

Reply:
no, proxy != instance of service in this case.  This function will be called in [cluster.go](https://github.com/istio/istio/pull/16657/files#diff-f507fb55b7edf4a03d3c7e9aebf64a1e), and the resulting policy passed to the helper functions you added in pilot/pkg/security [here](https://github.com/istio/istio/blob/master/pilot/pkg/security/authn/v1alpha1/model.go#L70).

At this point, `service` is one of the services in the mesh exposed to that particular `proxy`.  You are building outbound clusters for all the services in the mesh for that particular proxy/sidecar

Reply:
see comment below